human-readable-types/src/test/java/com/palantir/humanreadabletypes/StringBasedUrlTests.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.humanreadabletypes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+
+public final class StringBasedUrlTests {
+
+    @Test
+    public void canSerdeWithJackson() throws JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper();
+        StringBasedUrl url = mapper.readValue("\"http://foo\"", StringBasedUrl.class);
+        assertThat(url).isEqualTo(StringBasedUrl.valueOf("http://foo"));
+        assertThat(mapper.writeValueAsString(url)).isEqualTo("\"http://foo\"");
+    }
+
+    @Test
+    public void cannotConstructIllegalUri() {
+        assertThatThrownBy(() -> StringBasedUrl.valueOf(null)).isInstanceOf(IllegalArgumentException.class);
```

Comment:
nit: We tend to throw NPE in this case in most other places

Reply:
done